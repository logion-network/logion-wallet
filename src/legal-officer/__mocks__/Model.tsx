import {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
} from './ModelMock';

export {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
};

export function toRecoveryRequestType(value: string | undefined) {
    return value;
}
