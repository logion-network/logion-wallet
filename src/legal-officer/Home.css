.LegalOfficerHome .reading-container .Reading,
.LegalOfficerHome .reading-container button {
    margin-left: auto;
    margin-right: auto;
}

.LegalOfficerHome .reading-container .Reading {
    margin-top: -60px;
}

.LegalOfficerHome .reading-container button {
    display: block;
}

.LegalOfficerHome .transactions,
.LegalOfficerHome .collections {
    margin-top: 30px;
}

.LegalOfficerHome .transactions-button,
.LegalOfficerHome .identities-button,
.LegalOfficerHome .collections-button {
    position: absolute;
    top: 25px;
    right: 25px;
}

.LegalOfficerHome .identities {
    position: relative;
    margin-top: 30px;
}

@media (max-width: 1350px) {
    .LegalOfficerHome .reading-container .Reading {
        margin-top: 0;
    }
}
