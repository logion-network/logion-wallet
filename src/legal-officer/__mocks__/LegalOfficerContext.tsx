import {
    useLegalOfficerContext,
} from "./LegalOfficerContextMock";

export {
    useLegalOfficerContext
}
