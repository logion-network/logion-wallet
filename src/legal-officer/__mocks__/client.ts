import {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
} from "./ClientMock";

export {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
}
