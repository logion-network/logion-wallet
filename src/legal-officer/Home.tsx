import { Col, Row } from 'react-bootstrap';
import { useNavigate } from 'react-router';
import { Lgnt } from '@logion/node-api';
import { useLegalOfficerContext } from "./LegalOfficerContext";
import { useCommonContext } from "../common/CommonContext";
import { FullWidthPane } from '../common/Dashboard';
import Icon from '../common/Icon';
import Frame from '../common/Frame';
import Loader from '../common/Loader';
import Table, { ActionCell, Cell, DateCell, DateTimeCell, EmptyTableMessage } from '../common/Table';
import TransferAmountCell, { transferBalance } from '../common/TransferAmountCell';
import AmountCell from '../common/AmountCell';
import Reading from '../common/Reading';
import Button from '../common/Button';
import LocStatusCell from '../common/LocStatusCell';
import ButtonGroup from '../common/ButtonGroup';


import { dataLocDetailsPath, identityLocDetailsPath, locDetailsPath, locRequestsPath, transactionsPath } from './LegalOfficerPaths';

import './Home.css';
import UserIdentityNameCell from '../common/UserIdentityNameCell';
import { TransactionStatusCell } from "../common/TransactionStatusCell";
import { useLogionChain } from '../logion-chain';
import TransactionType from 'src/common/TransactionType';
import { toFeesClass } from "@logion/client/dist/Fees.js";

const MAX_WAITING_LOCS = 3;

export default function Account() {
    const { accounts } = useLogionChain();
    const {
        colorTheme,
        balanceState,
    } = useCommonContext();
    const { locs, locsState } = useLegalOfficerContext();
    const navigate = useNavigate();

    if (!balanceState || accounts === null || locsState === null || locsState.discarded) {
        return <Loader />;
    }

    return (
        <FullWidthPane
            mainTitle="Home"
            titleIcon={ {
                icon: {
                    id: 'home'
                },
                background: colorTheme.topMenuItems.iconGradient,
            } }
            className="LegalOfficerHome"
        >
            <Row>
                <Col>
                    <Frame
                        title="Last operation and balance"
                    >
                        <Row>
                            <Col>
                                <Table
                                    columns={ [
                                        {
                                            header: "Status",
                                            render: transaction => <TransactionStatusCell
                                                transaction={ transaction } />,
                                            width: "100px",
                                        },
                                        {
                                            header: "Transaction date",
                                            render: transaction => <DateCell dateTime={ transaction.createdOn } />,
                                            width: "200px",
                                        },
                                        {
                                            header: "Transaction type",
                                            render: transaction => <TransactionType address={ accounts.current!.accountId.address } transaction={ transaction } walletType="Wallet" />,
                                        },
                                        {
                                            header: "Amount",
                                            render: transaction => <TransferAmountCell
                                                amount={ transferBalance(accounts!.current!.accountId.address, transaction) } />,
                                            align: 'right',
                                            width: "120px",
                                        },
                                        {
                                            header: "Paid fees",
                                            render: transaction => <AmountCell
                                                amount={ toFeesClass(transaction.fees)?.totalFee || null } />,
                                            align: 'right',
                                            width: "120px",
                                        }
                                    ] }
                                    data={ balanceState.transactions.slice(0, 1) }
                                    renderEmpty={ () => <EmptyTableMessage>No transaction yet</EmptyTableMessage> }
                                />
                            </Col>
                            <Col>
                                <div className="reading-container">
                                    <Reading
                                        readingIntegerPart={ balanceState.balances[0].available.coefficient.toInteger() }
                                        readingDecimalPart={ balanceState.balances[0].available.coefficient.toFixedPrecisionDecimals(2) }
                                        unit={ balanceState.balances[0].available.prefix.symbol + Lgnt.CODE }
                                        level={ balanceState.balances[0].level }
                                    />
                                    <Button
                                        onClick={ () => navigate(transactionsPath("lgnt")) }
                                        slim={ true }
                                    >
                                        <Icon icon={ { id: "wallet" } } /> Go to my wallet
                                    </Button>
                                </div>
                            </Col>
                        </Row>
                    </Frame>

                    <Frame
                        className="identities"
                        title="Last waiting Identity LOC"
                    >
                        <Button
                            onClick={ () => navigate(locRequestsPath('Identity')) }
                            slim={ true }
                            className="identities-button"
                        >
                            <Icon icon={ { id: "identity" } } /> Go to identities
                        </Button>
                        <Table
                            columns={ [
                                {
                                    "header": "Requester",
                                    render: requestAndLoc => <UserIdentityNameCell
                                        userIdentity={ requestAndLoc.data().userIdentity } />,
                                    align: 'left',
                                },
                                {
                                    "header": "Description",
                                    render: requestAndLoc => <Cell content={ requestAndLoc.data().description }
                                                                   overflowing
                                                                   tooltipId='open-id-loc-description-tooltip' />,
                                    align: 'left',
                                },
                                {
                                    header: "Status",
                                    render: requestAndLoc => <LocStatusCell status={ requestAndLoc.data().status } />,
                                    width: "140px",
                                },
                                {
                                    "header": "Creation date",
                                    render: requestAndLoc => <DateTimeCell
                                        dateTime={ requestAndLoc.data().createdOn || null } />,
                                    width: '200px',
                                    align: 'center',
                                },
                                {
                                    header: "Action",
                                    render: requestAndLoc =>
                                        <ActionCell>
                                            <ButtonGroup>
                                                <Button
                                                    onClick={ () => navigate(identityLocDetailsPath(requestAndLoc.data().id.toString())) }>Manage
                                                    LOC</Button>
                                            </ButtonGroup>
                                        </ActionCell>
                                    ,
                                    width: '200px',
                                    align: 'center',
                                }
                            ] }
                            data={ locs["Identity"].workInProgress.slice(0, MAX_WAITING_LOCS) }
                            renderEmpty={ () => <EmptyTableMessage>No waiting LOC yet</EmptyTableMessage> }
                        />
                    </Frame>

                    <Frame
                        className="collections"
                        title="Last waiting Collection LOCs"
                    >
                        <Button
                            onClick={ () => navigate(locRequestsPath('Collection')) }
                            slim={ true }
                            className="collections-button"
                        >
                            <Icon icon={ { id: "loc" } } /> Go to collections
                        </Button>
                        <Table
                            columns={ [
                                {
                                    "header": "Requester",
                                    render: requestAndLoc => <UserIdentityNameCell
                                        userIdentity={ requestAndLoc.data().userIdentity } />,
                                    align: 'left',
                                },
                                {
                                    "header": "Description",
                                    render: requestAndLoc => <Cell content={ requestAndLoc.data().description }
                                                                   overflowing
                                                                   tooltipId='open-id-loc-description-tooltip' />,
                                    align: 'left',
                                },
                                {
                                    header: "Status",
                                    render: requestAndLoc => <LocStatusCell status={ requestAndLoc.data().status } />,
                                    width: "140px",
                                },
                                {
                                    "header": "Creation date",
                                    render: requestAndLoc => <DateTimeCell
                                        dateTime={ requestAndLoc.data().createdOn || null } />,
                                    width: '200px',
                                    align: 'center',
                                },
                                {
                                    header: "Action",
                                    render: requestAndLoc =>
                                        <ActionCell>
                                            <ButtonGroup>
                                                <Button
                                                    onClick={ () => navigate(locDetailsPath(requestAndLoc.data().id.toString(), "Collection")) }>Manage
                                                    LOC</Button>
                                            </ButtonGroup>
                                        </ActionCell>
                                    ,
                                    width: '200px',
                                    align: 'center',
                                }
                            ] }
                            data={ locs["Collection"].workInProgress.slice(0, MAX_WAITING_LOCS) }
                            renderEmpty={ () => <EmptyTableMessage>No waiting LOC yet</EmptyTableMessage> }
                        />
                    </Frame>

                    <Frame
                        className="transactions"
                        title="Last waiting Transaction LOC"
                    >
                        <Button
                            onClick={ () => navigate(locRequestsPath('Transaction')) }
                            slim={ true }
                            className="transactions-button"
                        >
                            <Icon icon={ { id: "loc" } } /> Go to transactions
                        </Button>
                        <Table
                            columns={ [
                                {
                                    "header": "Requester",
                                    render: requestAndLoc => <UserIdentityNameCell
                                        userIdentity={ requestAndLoc.data().userIdentity } />,
                                    align: 'left',
                                },
                                {
                                    "header": "Description",
                                    render: requestAndLoc => <Cell content={ requestAndLoc.data().description }
                                                                   overflowing
                                                                   tooltipId='open-loc-description-tooltip' />,
                                    align: 'left',
                                },
                                {
                                    header: "Status",
                                    render: requestAndLoc => <LocStatusCell status={ requestAndLoc.data().status } />,
                                    width: "140px",
                                },
                                {
                                    "header": "Creation date",
                                    render: requestAndLoc => <DateTimeCell
                                        dateTime={ requestAndLoc.data().createdOn || null } />,
                                    width: '200px',
                                    align: 'center',
                                },
                                {
                                    header: "Action",
                                    render: requestAndLoc =>
                                        <ActionCell>
                                            <ButtonGroup>
                                                <Button
                                                    onClick={ () => navigate(dataLocDetailsPath(requestAndLoc.data().locType, requestAndLoc.data().id.toString())) }>Manage
                                                    LOC</Button>
                                            </ButtonGroup>
                                        </ActionCell>
                                    ,
                                    width: '200px',
                                    align: 'center',
                                }
                            ] }
                            data={ locs['Transaction'].workInProgress.slice(0, MAX_WAITING_LOCS) }
                            renderEmpty={ () => <EmptyTableMessage>No waiting LOC yet</EmptyTableMessage> }
                        />
                    </Frame>
                </Col>
            </Row>
        </FullWidthPane>
    );
}
