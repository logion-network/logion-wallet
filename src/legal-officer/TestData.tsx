import { DateTime } from 'luxon';
import { toIsoString } from "@logion/client";
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient.js';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
