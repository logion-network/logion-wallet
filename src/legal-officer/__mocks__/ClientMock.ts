import { OpenLoc, VerifiedThirdParty } from "@logion/client";

export let closeLoc = jest.fn().mockResolvedValue(undefined);

export function setCloseLocMock(mock: any) {
    closeLoc = mock;
}

export let deleteLink = jest.fn().mockResolvedValue(undefined);

export let voidLoc = jest.fn().mockResolvedValue(undefined);

export function setVoidLocMock(mock: any) {
    voidLoc = mock;
}

let vtpSelections: VerifiedThirdParty[] = [];

export async function getVerifiedThirdPartySelections(params: { locState: OpenLoc } ): Promise<VerifiedThirdParty[]> {
    return Promise.resolve(vtpSelections);
}

export function setVerifiedThirdPartySelections(mock: VerifiedThirdParty[]) {
    vtpSelections = mock;
}

export let requestVote = jest.fn().mockResolvedValue(undefined);

export function setRequestVoteMock(mock: any) {
    requestVote = mock;
}
