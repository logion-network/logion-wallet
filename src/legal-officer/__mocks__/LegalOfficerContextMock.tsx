import { LocType, IdentityLocType } from "@logion/node-api/dist/Types";
import { AxiosInstance } from "axios";

import { COLOR_THEME } from "src/common/TestData";
import { RequestAndLoc } from "../LegalOfficerContext";
import { LegalOfficerData } from "../LegalOfficerData";

export let pendingTokenizationRequests: any[] | null = null;

export let rejectRequest = () => {

};

export let tokenizationRequestsHistory: any[] | null = null;

export let refreshRequests = jest.fn();

export let pendingProtectionRequests: any[] | null = null;

export let activatedProtectionRequests: any[] | null = null;

export let protectionRequestsHistory: any[] | null = null;

export let pendingRecoveryRequests: any[] | null = null;

export let recoveryRequestsHistory: any[] | null = null;

export function setPendingRequests(requests: any[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectRequest(callback: any) {
    rejectRequest = callback;
}

export function setTokenizationRequestsHistory(requests: any[]) {
    tokenizationRequestsHistory = requests;
}

export function setRefreshRequests(callback: any) {
    refreshRequests = callback;
}

export function setPendingProtectionRequests(requests: any[]) {
    pendingProtectionRequests = requests;
}

export function setActivatedProtectionRequests(requests: any[]) {
    activatedProtectionRequests = requests;
}

export function setProtectionRequestsHistory(requests: any[]) {
    protectionRequestsHistory = requests;
}

export function setPendingRecoveryRequests(requests: any[]) {
    pendingRecoveryRequests = requests;
}

export function setRecoveryRequestsHistory(requests: any[]) {
    recoveryRequestsHistory = requests;
}

export let pendingLocRequests: Record<LocType, any[]> | null = null;
export let rejectedLocRequests: Record<LocType, any[]> | null = null;

export let openedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let closedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let openedIdentityLocs: RequestAndLoc[] | null = null;

export let openedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let closedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let voidIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export function setRejectedLocRequests(requests: any[]) {
    rejectedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedLocRequests(requests: any[]) {
    openedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setPendingLocRequests(requests: any[]) {
    pendingLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setClosedLocRequests(requests: any[]) {
    closedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedIdentityLocs(requests: any[]) {
    openedIdentityLocs = requests;
}

export function setOpenedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    openedIdentityLocsByType = locs;
}

export function setClosedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    closedIdentityLocsByType = locs;
}

export function setVoidedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    voidIdentityLocsByType = locs;
}

let onchainSettings: LegalOfficerData = {};

export function setOnchainSettings(settings: LegalOfficerData) {
    onchainSettings = settings;
}

export let refreshLocs = jest.fn();

export function useLegalOfficerContext() {
    return {
        pendingTokenizationRequests,
        rejectRequest,
        tokenizationRequestsHistory,
        refreshRequests,
        pendingProtectionRequests,
        activatedProtectionRequests,
        protectionRequestsHistory,
        colorTheme: COLOR_THEME,
        openedLocRequests,
        openedIdentityLocs,
        openedIdentityLocsByType,
        closedLocRequests,
        closedIdentityLocsByType,
        pendingLocRequests,
        rejectedLocRequests,
        voidIdentityLocsByType,
        axios: {} as AxiosInstance,
        onchainSettings,
        pendingRecoveryRequests,
        refreshLocs,
    };
}
