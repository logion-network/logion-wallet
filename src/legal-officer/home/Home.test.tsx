jest.mock("react-router");
jest.mock("../../common/CommonContext");
jest.mock("../../logion-chain");
jest.mock('../LegalOfficerContext');

import { BalanceState } from "@logion/client/dist/Balance.js";
import { DEFAULT_COIN_BALANCE, DEFAULT_TRANSACTION } from "../../common/TestData";
import { setBalanceState, } from "../../common/__mocks__/CommonContextMock";
import { shallowRender } from "../../tests";
import Home from "./Home";

describe("Home", () => {

    it("renders", () => {
        setBalanceState({
            balance: DEFAULT_COIN_BALANCE,
            transactions: [ DEFAULT_TRANSACTION ],
        } as BalanceState);
        const tree = shallowRender(<Home />)
        expect(tree).toMatchSnapshot();
    });
});
