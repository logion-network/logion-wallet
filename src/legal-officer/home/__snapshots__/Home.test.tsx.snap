// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<FullWidthPane
  className="LegalOfficerHome"
  mainTitle="Home"
  titleIcon={
    Object {
      "background": Object {
        "from": "#3b6cf4",
        "to": "#6050dc",
      },
      "icon": Object {
        "id": "home",
      },
    }
  }
>
  <Row>
    <Col>
      <Frame
        title="Last operation and balance"
      >
        <Row>
          <Col>
            <Table
              columns={
                Array [
                  Object {
                    "header": "Status",
                    "render": [Function],
                    "width": "100px",
                  },
                  Object {
                    "header": "Transaction date",
                    "render": [Function],
                    "width": "200px",
                  },
                  Object {
                    "header": "Transaction type",
                    "render": [Function],
                  },
                  Object {
                    "align": "right",
                    "header": "Amount",
                    "render": [Function],
                    "width": "120px",
                  },
                  Object {
                    "align": "right",
                    "header": "Paid fees",
                    "render": [Function],
                    "width": "120px",
                  },
                ]
              }
              data={
                Array [
                  Object {
                    "createdOn": "2021-07-28T12:30:00.000",
                    "fees": Object {
                      "inclusion": "125000",
                      "total": "125000",
                    },
                    "from": ValidAccountId {
                      "anyAccountId": AnyAccountId {
                        "address": "vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz",
                        "type": "Polkadot",
                      },
                    },
                    "id": "id",
                    "method": "method",
                    "pallet": "pallet",
                    "reserved": "0",
                    "successful": true,
                    "tip": "0",
                    "to": ValidAccountId {
                      "anyAccountId": AnyAccountId {
                        "address": "vQuQm7K3Qt1nQYDzqQzCPQj6b4AQFPZgD5FLnEpBTvCx1rVix",
                        "type": "Polkadot",
                      },
                    },
                    "total": "125420",
                    "transferDirection": "Sent",
                    "transferValue": "420",
                    "type": "EXTRINSIC",
                  },
                ]
              }
              renderEmpty={[Function]}
            />
          </Col>
          <Col>
            <div
              className="reading-container"
            >
              <Reading
                level={0.1}
                readingDecimalPart="00"
                readingIntegerPart="42"
                unit="aLGNT"
              />
              <Button
                onClick={[Function]}
                slim={true}
              >
                <Icon
                  icon={
                    Object {
                      "id": "wallet",
                    }
                  }
                />
                 Go to my wallet
              </Button>
            </div>
          </Col>
        </Row>
      </Frame>
      <Frame
        className="identities"
        title="Last waiting Identity LOC"
      >
        <Button
          className="identities-button"
          onClick={[Function]}
          slim={true}
        >
          <Icon
            icon={
              Object {
                "id": "identity",
              }
            }
          />
           Go to identities
        </Button>
        <Table
          columns={
            Array [
              Object {
                "align": "left",
                "header": "Requester",
                "render": [Function],
              },
              Object {
                "align": "left",
                "header": "Description",
                "render": [Function],
              },
              Object {
                "header": "Status",
                "render": [Function],
                "width": "140px",
              },
              Object {
                "align": "center",
                "header": "Creation date",
                "render": [Function],
                "width": "200px",
              },
              Object {
                "align": "center",
                "header": "Action",
                "render": [Function],
                "width": "200px",
              },
            ]
          }
          data={Array []}
          renderEmpty={[Function]}
        />
      </Frame>
      <Frame
        className="collections"
        title="Last waiting Collection LOCs"
      >
        <Button
          className="collections-button"
          onClick={[Function]}
          slim={true}
        >
          <Icon
            icon={
              Object {
                "id": "loc",
              }
            }
          />
           Go to collections
        </Button>
        <Table
          columns={
            Array [
              Object {
                "align": "left",
                "header": "Requester",
                "render": [Function],
              },
              Object {
                "align": "left",
                "header": "Description",
                "render": [Function],
              },
              Object {
                "header": "Status",
                "render": [Function],
                "width": "140px",
              },
              Object {
                "align": "center",
                "header": "Creation date",
                "render": [Function],
                "width": "200px",
              },
              Object {
                "align": "center",
                "header": "Action",
                "render": [Function],
                "width": "200px",
              },
            ]
          }
          data={Array []}
          renderEmpty={[Function]}
        />
      </Frame>
      <Frame
        className="transactions"
        title="Last waiting Transaction LOC"
      >
        <Button
          className="transactions-button"
          onClick={[Function]}
          slim={true}
        >
          <Icon
            icon={
              Object {
                "id": "loc",
              }
            }
          />
           Go to transactions
        </Button>
        <Table
          columns={
            Array [
              Object {
                "align": "left",
                "header": "Requester",
                "render": [Function],
              },
              Object {
                "align": "left",
                "header": "Description",
                "render": [Function],
              },
              Object {
                "header": "Status",
                "render": [Function],
                "width": "140px",
              },
              Object {
                "align": "center",
                "header": "Creation date",
                "render": [Function],
                "width": "200px",
              },
              Object {
                "align": "center",
                "header": "Action",
                "render": [Function],
                "width": "200px",
              },
            ]
          }
          data={Array []}
          renderEmpty={[Function]}
        />
      </Frame>
      <Frame
        className="withdrawals"
        title="Last waiting vault withdrawals"
      >
        <Button
          className="withdrawals-button"
          onClick={[Function]}
          slim={true}
        >
          <Icon
            icon={
              Object {
                "id": "vault",
              }
            }
          />
           Go to vault withdrawals
        </Button>
        <Table
          columns={
            Array [
              Object {
                "align": "left",
                "header": "User",
                "render": [Function],
                "renderDetails": [Function],
              },
              Object {
                "header": "Type",
                "render": [Function],
                "width": "80px",
              },
              Object {
                "align": "right",
                "header": "Amount",
                "render": [Function],
                "width": "120px",
              },
              Object {
                "header": "Status",
                "render": [Function],
                "width": "150px",
              },
              Object {
                "header": "Creation date",
                "render": [Function],
                "width": "130px",
              },
            ]
          }
          data={Array []}
          renderEmpty={[Function]}
        />
      </Frame>
    </Col>
  </Row>
</FullWidthPane>
`;
